### DISCUSSION TITLE
Recommended Quarkus OpenAPI support: contract first vs code first

### DISCUSSION THREAD

**Discussion by ppatierno** [quarkusio/quarkus#32537] [Q&A]
Hi,
I am working on porting the Strimzi Kafka bridge project to use Quarkus and the HTTP REST part is already implemented using RESTeasy Reactive.
What I have to deal with right now is OpenAPI and our current implementation is using "contract first" approach.
An OpenAPI spec file is provided and the Vert.x OpenAPI component uses it to help creating the routes and to validate the HTTP requests.
While moving to Quarkus, I haven't see a kind of "official" way to use a "contract first" approach.
The official doc mentions [using Swagger](https://quarkus.io/guides/openapi-swaggerui) so with using annotation on the REST API methods in order to generate and serve the OpenAPI spec.
On the other side, I came into this extension [Quarkus OpenAPI Generator](https://github.com/quarkiverse/quarkus-openapi-generator) which is anyway based on a project not well supported imho and also it mentions:

> This extension is for REST code generation for client side only. If you're looking for code generation for the server side, please take a look at the [Quarkus Apicurio Extension](https://github.com/Apicurio/apicurio-codegen/tree/main/quarkus-extension).

So I followed that Quarkus Apicurio Extension but it doesn't seem to be in the extentions catalog yet, or I wasn't able to pick it :-(

In summary, what's the recommended way to have OpenAPI support with Quarkus?

[a comment by an automated account is not shown]

**Comment by maxandersen** [comment]
Hmm. Seems like a mistake it's not available. At least as preview is best I expected.
